packages/core/admin/server/src/controllers/admin.ts

```diff
@@ -12,8 +16,23 @@ import {
 } from '../validation/project-settings';
 import { getService } from '../utils';
 
+import type {
+  Init,
+  ProjectSettings,
+  Information,
+  Plugins,
+  TelemetryProperties,
+} from '../../../shared/contracts/admin';
+
 const { isUsingTypeScript } = tsUtils;
 
+// TODO what is the best way to extend the Request interface with files?
```

Comment:
wdyt here @Marc-Roig ? the requests contain files in many cases

Reply:
this is interesting! Good idea to extend it like thtat. I wonder if we should add this directly into strapi types, but this seems like a great idea in the meantime.

Reply:
another less mutating approach could be creating another Context type that extends the original, using the Replace util that will soon be merged:

```ts
type FileRequest = Context['request'] & { file: any }
type FileContext = Replace<Context & { request: FileRequest}>
```

Reply:
Thanks! I've added a TODO so we can extend the request like this once Replace<> is available